### PR TITLE
docs: update plugin concept link in README

### LINKED ISSUE #6880: docs: change the url of plugin concept in README
### Current State

**Current state of the README file**

Reading [plugin concept](https://github.com/apache/apisix/blob/master/docs/en/latest/architecture-design/plugin.md) would help you learn more about the plugin.

**Redirected Page**

404 page

### Desired State

**Desired state of the README file**

Reading [plugin concept](https://github.com/apache/apisix/blob/master/docs/en/latest/architecture-design/apisix.md) would help you learn more about the plugin.

**Redirected Page**

desiredPage

### PR DESCRIPTION
**Pull request by gdsghost**
### Description

Fixes #6880

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -184,7 +184,7 @@ A/B testing, canary release, blue-green deployment, limit rate, defense against
 4. Plugin development
 
    You can refer to [plugin development guide](docs/en/latest/plugin-develop.md), and sample plugin `example-plugin`'s code implementation.
-   Reading [plugin concept](docs/en/latest/architecture-design/plugin.md) would help you learn more about the plugin.
+   Reading [plugin concept](docs/en/latest/architecture-design/apisix.md) would help you learn more about the plugin.
```
Comment (hf400159) [inline README.md:187]:
```suggestion
   Reading [plugin concept](docs/en/latest/terminology/plugin.md) would help you learn more about the plugin.
```

Comment (hf400159) [reply README.md:187]: `plugin.md`  has moved to `terminology`.

Comment (spacewander) [reply README.md:187]: Could we add some link linter?
CC @navendu-pottekkat

Comment (navendu-pottekkat) [reply README.md:187]: We can add link linter but since pages are across multiple repos, it would be difficult to do it statically.

Comment (gdsghost) [reply README.md:187]: Can't we add linter in github workflows. 
May I know the meaning of availability of pages on multiple repos?
CC @navendu-pottekkat

Comment (navendu-pottekkat) [reply README.md:187]: I did not realise that this was a change in README. For this case, a linter in GitHub workflows should work. There should be some action available that we can use.

For APISIX documentation, we have multiple repos and documentation for each of the components are in different repos. This is pulled in the apisix-website repo during a build. So, pages could be linked across repos and I don't think we can have static lint checks to catch them. We may need to build the website and test the links from the built website using some workflow.

Comment (SkyeYoung) [reply README.md:187]: > I did not realise that this was a change in README. For this case, a linter in GitHub workflows should work. There should be some action available that we can use.
> 
> For APISIX documentation, we have multiple repos and documentation for each of the components are in different repos. This is pulled in the apisix-website repo during a build. So, pages could be linked across repos and I don't think we can have static lint checks to catch them. We may need to build the website and test the links from the built website using some workflow.

@navendu-pottekkat  We already have one, pls see https://github.com/apache/apisix-website/issues/1047.

**Review by spacewander — APPROVED** [review]

**Review by navendu-pottekkat — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**